### PR TITLE
Fix export

### PR DESCRIPTION
**Pull request by t-h2o**
#### New
```sh
it's the Prompt $ export a
it's the Prompt $ env
HOME=myhome
a
it's the Prompt $ export a=1
it's the Prompt $ env
HOME=myhome
a=1
```
* Replace variable
* Set variable without `=`

### CODE REVIEW

File: srcs/export.c [hunk 1]
```diff
@@ -6,70 +6,75 @@
...
+	i = 0;
+	while (envp[i])
+		++i;
+	if (pos == -1)
 	{
-		if (ind[2])
-			ret[ind[2]] = table[ind[2]];
-		else
-			ret[0] = str_dup(table[0]);
+		new_env = malloc((i + 1 + 1) * sizeof(char *));
```
Comment (ldominiq) [inline srcs/export.c:46]: malloc not protected >:(

File: srcs/export.c [hunk 2]
```diff
@@ -6,70 +6,75 @@
...
+		new_env[i] = str_dup(add);
+		while (i-- > 0)
+			new_env[i] = envp[i];
 	}
-	free(*table);
-	free(table);
-	return (ret);
+	else
+	{
+		new_env = malloc((i + 1) * sizeof(char *));
```
Comment (ldominiq) [inline srcs/export.c:54]: malloc not protected >:(

**Review by ldominiq — CHANGES_REQUESTED** [review]

File: srcs/export.c (hunk 1 above)
Comment (t-h2o) [reply srcs/export.c:46]: No risk no fun

File: srcs/export.c (hunk 2 above)
Comment (t-h2o) [reply srcs/export.c:54]: no RISKKK no FUNNNNNNNNNN

**Pull request merged**